### PR TITLE
Purchases: Add `excluding taxes` string on purchase screens

### PR DESCRIPTION
**Pull request by JessBoctor**
Related to https://github.com/Automattic/payments-shilling/issues/2184

## Proposed Changes

Outside the USA, it is not often assumed that displayed prices are exclusive of tax. This can be a jarring experience when a customer gets to checkout and has an increase to their total because taxes are added. 

To help mitigate this experience, all amounts within the `me/purchases` section of Calypso should explicitly state whether they are exclusive or inclusive of taxes. Billing receipts already display any tax amounts within the `(incl. $xx.xx tax)` strings. 

This PR appends a string stating `(excl. tax)` to any amounts shown on the purchases list or an individual purchase management screen. The `<abbr>` tag surrounding the string has a `title` added which acts as a tool-tip when hovered over. 

**Before**

Screenshot 2024-01-16 at 12 36 25

Screenshot 2024-01-16 at 12 36 40

**AFTER**

Screenshot 2024-01-18 at 12 32 02

Screenshot 2024-01-18 at 12 32 37

Screenshot 2024-01-16 at 11 12 13

Screenshot 2024-01-16 at 11 12 30

Currently, the country information for the purchase is not included in these screens (or at least I am not sure how to access it). Only the locale information is available. This means we cannot yet utilize the official tax names for a country (e.g. "VAT" for the EU, "CT" for Japan, etc). We are relying on the Locale translation of `tax` or the fallback of `tax (VAT/GST/CT)` instead.

## Testing Instructions

* Apply this patch and visit `me/purchases`. Any purchases which have a renewal amount should have ` (excl. tax)` appended to the amount. 
* From `me/purchases` click on a purchase with a renewal amount. On the purchase management screen, you should see `excludes tax` or `excludes tax (VAT/GST/CT)` under the renewal amount.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/me/purchases/manage-purchase/purchase-meta.tsx [hunk 1]
```diff
@@ -81,6 +81,27 @@ export default function PurchaseMeta( {
...
+			: translate( 'tax (VAT/GST/CT)' );
+
+	/* translators: %s is the name of taxes in the country (eg: "VAT" or "GST"). */
+	const excludeTaxStringAbbrevation = translate( 'excludes %s', {
+		textOnly: true,
+		args: [ taxName ],
+	} );
+
+	/* translators: %s is the name of taxes in the country (eg: "VAT" or "GST"). */
+	const excludeTaxStringTitle = translate( 'Renewal price exlcudes any applicable %s', {
```
Comment (jjchrisdiehl) [inline client/me/purchases/manage-purchase/purchase-meta.tsx]: Typo 'exlcudes'

File: client/me/purchases/purchase-item/index.jsx [hunk 2]
```diff
@@ -66,6 +66,27 @@ class PurchaseItem extends Component {
 	getStatus() {
 		const { purchase, translate, locale, moment, name, isJetpack, isDisconnectedSite } = this.props;
 		const expiry = moment( purchase.expiryDate );
+		// To-do: There isn't currently a way to get the taxName based on the country.
```
Comment (jjchrisdiehl) [inline client/me/purchases/purchase-item/index.jsx:70]: Since this is being used in two places, maybe we can move it to a helper function file such as `client/my-sites/purchases/utils.js` though I suppose this work could be helpful in checkout too, but there isn't a common parent directory where we could nest it 🤔

**Comment by jjchrisdiehl** [comment]
Functional testing:

Works well, minus the noted typo
image

Here too
image

I don't think these strings are translated yet, so you'll also need to add the `String Freeze` tag to your PR and wait for the automated translation process to finish.

Great work!

File: client/me/purchases/purchase-item/index.jsx [hunk 3]
```diff
@@ -66,6 +66,27 @@ class PurchaseItem extends Component {
...
+		// The country is not includedin the purchase information envelope
+		// We should add this information so we can utilize useTaxName to retrieve the correct taxName
```
Comment (sirbrillig) [inline client/me/purchases/purchase-item/index.jsx:72]: Interesting! Receipts (eg: from the endpoint `/me/billing-history/past` or the single receipt endpoint) do include this information with the `tax_country_code` property, but "purchases" (which I guess means subscriptions, from the endpoint `/sites/188830194/purchases`) do not. Since each subscription should have an associated receipt, we should be able to modify the endpoint to include the tax data. But yes, it could be done as a follow-up to this change.

File: client/me/purchases/purchase-item/index.jsx [hunk 4]
```diff
@@ -66,6 +66,27 @@ class PurchaseItem extends Component {
...
+		// For now, we are using a fallback tax name
+		const taxName =
+			getLocaleSlug()?.startsWith( 'en' ) || i18n.hasTranslation( 'tax' )
+				? translate( 'tax' )
+				: translate( 'tax (VAT/GST/CT)' );
+
+		/* translators: "excl."" is an abbrevation of the word "exclude" */
+		/* translators: %s is the name of taxes in the country (eg: "VAT" or "GST"). */
+		const excludeTaxStringAbbrevation = translate( '(excl. %s)', {
```
Comment (sirbrillig) [inline client/me/purchases/purchase-item/index.jsx]: What's the reason for using `excl.` instead of `excludes`? If it's to save space, that won't help because in other languages there might not be a conveniently sized abbreviation. We must assume that all words will be very long and make sure the UI adapts.

File: client/me/purchases/manage-purchase/purchase-meta.tsx [hunk 5]
```diff
@@ -91,6 +112,9 @@ export default function PurchaseMeta( {
 						<PurchaseMetaPrice purchase={ purchase } />
 						<PurchaseMetaIntroductoryOfferDetail purchase={ purchase } />
 					</span>
+					<span>
+						<abbr title={ excludeTaxStringTitle }>{ excludeTaxStringAbbrevation }</abbr>
```
Comment (sirbrillig) [inline client/me/purchases/manage-purchase/purchase-meta.tsx]: I'm not certain but I think `abbr` tags may not work on touch devices. At least, I don't seem to be able to see their `title` content on my iPhone. 🤔

File: client/me/purchases/purchase-item/index.jsx (hunk 4 above)
Comment (JessBoctor) [reply client/me/purchases/purchase-item/index.jsx]: Is following instructions a good enough reason? I am not attached to it and will update.

File: client/me/purchases/manage-purchase/purchase-meta.tsx (hunk 1 above)
Comment (JessBoctor) [reply client/me/purchases/manage-purchase/purchase-meta.tsx]: Fixed and added a spell checker to VSCode.

File: client/me/purchases/purchase-item/index.jsx (hunk 4 above)
Comment (JessBoctor) [reply client/me/purchases/purchase-item/index.jsx]: Resolved this conversation in p1705599023042179-Slack-C0117V2PCAE/

tl;dr I am changing `(excl. tax)` to `(excludes tax)`. This should be clear enough for all users. We'll go ahead and leave the `<abbr>` title property in place for extra clarity.

File: client/me/purchases/purchase-item/index.jsx (hunk 3 above)
Comment (JessBoctor) [reply client/me/purchases/purchase-item/index.jsx:72]: 👍  I'll make an issue for it.

**Review by sirbrillig — APPROVED** [review]
Works as described!

Screenshot 2024-01-18 at 1 35 18 PM

Screenshot 2024-01-18 at 1 35 41 PM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10846522

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @JessBoctor for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10846522).
